gt/main.c

```diff
@@ -1466,7 +1466,7 @@ process_gt_cmd(struct gt_cmd_entry *entry, struct gt_instance *instance)
 					!!entry->u.bc.is_returned, 0) != 0)) {
 			GT_LOG(ERR, "gt: failed to incrementally update lua state at lcore %u: %s\n",
 				rte_lcore_id(),
-				luaL_checkstring(instance->lua_state, -1));
+				lua_tostring(instance->lua_state, -1));
```

Comment:
What does `GT_LOG()` print when it receives `NULL` for `%s`?

Reply:
Under the hood, `GT_LOG()` is using `vfprintf()`, for which I did not find any defined behavior for passing `NULL` for `%s`.

But it's worth noting that this code is being called only when either `luaL_loadbuffer()` or `lua_pcall()` fails, both of which promise to push an error string onto the stack when they fail. In `config/dynamic.c` (where we just agreed to remove the `if (ret != 0) { ... }`), we indeed make the assumption that if one of these functions fails, that we can at least read the error message from the stack.

If we're not comfortable with this assumption, then there's at least a couple places we need to address it.

Reply:
Could you test calling `GT_LOG("%s")` with `NULL`? I think it properly handles this case.

Reply:
It replaces `%s` with the string `(null)`.

Reply:
That is that safe. Case closed. Thank you. :)